src/SubsMgr.cpp

```diff
@@ -409,6 +409,9 @@ void SubsMgr::maybeCacheStatusResult(const HashX &sh, const SubStatus &status)
     else if (auto *dsp = status.dsproof(); dsp && !dsp->isComplete() && !dsp->isEmpty())
         // we only allow empty (default constructred) DSProofs or ones that are isComplete(), otherwise reject
         return;
+    else if (auto *hs = status.hashSet(); hs && !(*hs)->size())
```

Comment:
Not sure if this is correct. So we want to cache only if the hashset is not empty it seems.